includes/specials/SpecialBlockList.php

```diff
@@ -431,12 +430,16 @@ function preprocessResults( $result ){
 			$lb->add( NS_USER_TALK, $name );
 		}
 
-		$ua = UserArray::newFromIDs( $userids );
-		foreach( $ua as $user ){
-			$name = str_replace( ' ', '_', $user->getName() );
-			$lb->add( NS_USER, $name );
-			$lb->add( NS_USER_TALK, $name );
-		} 
+		$result->rewind();
+
+		// SUS-3108: Use username lookup to fetch user names
+		$this->userNames = User::whoAre( $userIds );
+		foreach ( $this->userNames as $userId => $userName ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$userId" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app